### PR TITLE
Retry sending event message on timeout.

### PR DESCRIPTION
**Pull request by fiadliel**
This is similar to the behaviour of the honeycomb-go client, in that it retries once in case of  timeout.

### CODE REVIEW

File: src/Honeycomb/Api/Events.hs [hunk 1]
```diff
@@ -25,15 +25,21 @@ maxBodyLength :: Int64
...
   (Client.Request -> m (Client.Response LBS.ByteString)) ->
   RequestOptions ->
   [Event] ->
   m SendEventsResponse
 sendEvents httpLbs requestOptions events = do
-  initReq <- liftIO $ Client.parseRequest batchUri
+  req <- liftIO $ newReq <$> Client.parseRequest batchUri
   let (unsent, dropped, body) = createBodyFromEvents True events [] "[" 1
-  response <- httpLbs $ newReq initReq $ body <> "]"
+      reqWithBody = req body
```
Comment (andreyk0) [inline src/Honeycomb/Api/Events.hs]: I think it's a good idea for retries to be user-configurable (with sensible defaults, of course), including disabling retry functionality all together. How do you feel about something like https://www.stackage.org/haddock/lts-14.25/retry-0.8.1.0/Control-Retry.html ? It seems to have a data structure that captures "retry policy", could be something you set up / let users tweak during lib bootstrap and then just use.

Comment (fiadliel) [reply src/Honeycomb/Api/Events.hs]: I don't actually want this to be user-configurable; sensible values are more up to Honeycomb  than users of a client library (who would rather that HC's servers burn down in flames than that any data might actually get lost). And many clients with a retry policy can cause "thundering herd" failures upstream.

The retry policy should not be any more spammy than any of HC's official libraries, so this is as complex as it's going to get for now.

_If_ other libraries change their behaviour, then adding retry might make sense then.

Comment (fiadliel) [reply src/Honeycomb/Api/Events.hs]: Disabling retry functionality has marginal use, if you are really _really_ paranoid about duplicated events, and would prefer to drop those events instead. Seems like it's premature to add that feature at this point (no other library has the ability to customise this).

Comment (andreyk0) [reply src/Honeycomb/Api/Events.hs]: :+1:

**Pull request merged**